File: backend/src/main/java/space/davids_digital/sweetie/integration/vk/dto/VkSilentTokenExchangeResultDto.java
package space.davids_digital.sweetie.integration.vk.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class VkSilentTokenExchangeResultDto {
    @JsonProperty("access_token")
    public String accessToken;

    @JsonProperty("access_token_id")
    public String accessTokenId;

    @JsonProperty("user_id")
    public long userId;

    @JsonProperty("additional_signup_required")
    public boolean additionalSignupRequired;

    @JsonProperty("is_partial")
    public boolean isPartial;

    @JsonProperty("is_service")
    public boolean isService;

    @JsonProperty("source")
    public long source;

    @JsonProperty("source_description")
    public String sourceDescription;

    @JsonProperty("expires_in")
    public long expiresIn;
}

File: backend/src/main/java/space/davids_digital/sweetie/integration/vk/dto/VkSilentTokenExchangeResultDto.kt
package space.davids_digital.sweetie.integration.vk.dto

import com.fasterxml.jackson.annotation.JsonProperty

@Deprecated("Use new API")
open class VkSilentTokenExchangeResultDto {
    @JsonProperty("access_token")
    var accessToken: String? = null

    @JsonProperty("access_token_id")
    var accessTokenId: String? = null

    @JsonProperty("user_id")
    var userId: Long = 0

    @JsonProperty("additional_signup_required")
    var additionalSignupRequired = false

    @JsonProperty("is_partial")
    var isPartial = false

    @JsonProperty("is_service")
    var isService = false

    @JsonProperty("source")
    var source: Long = 0

    @JsonProperty("source_description")
    var sourceDescription: String? = null

    @JsonProperty("expires_in")
    var expiresIn: Long = 0
}
